lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -513,15 +390,14 @@ defmodule RealtimeWeb.RealtimeChannel do
     GenCounter.add(id)
 
     case RateCounter.get(id) do
-      {:ok, %{avg: avg}} ->
-        if avg < limits.max_joins_per_second do
-          :ok
-        else
-          {:error, :too_many_joins}
-        end
+      {:ok, %{avg: avg}} when avg < limits.max_joins_per_second ->
+        :ok
+
+      {:ok, %{avg: _}} ->
+        {:error, :too_many_joins}
 
       other ->
-        Logger.error("Unexpected error for #{tenant}: " <> inspect(other))
+        Logger.error("Unexpected error for " <> tenant <> ": " <> inspect(other))
```

Comment:
Could it be possible that the tenant is not binary here?

Reply:
from what I gather it seems it's possible, will add a safeguard

Reply:
actually will remove the tenant from there since we have that in the metadata of the logs